### PR TITLE
[MOB-2755] Hide progress bar if in overlay mode

### PR DESCRIPTION
**Pull request by lucaschifino**
[MOB-2755]

## Context

There is a visual bug after the upgrade where the progress bar is visible while typing (see ticket above).

### Root cause

We observe `estimatedProgress` and [update the bar when it changes](https://github.com/ecosia/ios-browser/blob/b2788392f378c09c6dd88d3a00507075be129a3e/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift#L1376-L1384), **independent of wether the overlay is showing** (overlay being what appear once the user clicks on the url bar).

### Solution

The simplest change to fix that was to keep the progress bar hidden if we are in overlay mode.

I investigated whether it could mean this is mistakenly kept hidden once user exits overlay mode, and, though this makes theoretical sense, I believe the change is quite safe give the next time the progress is update it will be shown again (unless it's done in which case we do want to keep it hidden).

### Checklist

- [x] I performed some relevant testing on a real device and/or simulator
- [x] I added the `// Ecosia:` helper comments where needed

[MOB-2755]: https://ecosia.atlassian.net/browse/MOB-2755?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by d4r1091 — APPROVED** [review]
Neat 👌 👏 
Non blocking comment: is this `isHidden` set elsewhere that would invalidate this scenario?

**Comment by lucaschifino** [comment]
> Non blocking comment: is this isHidden set elsewhere that would invalidate this scenario?

There are a couple places when overlay is initially prepared, that's why it is hidden as soon as the url bar is selected, though this gets overwritten as soon as `estimatedProgress` changes using exactly the method that I updated. 

TLDR: Only in different flows so we should be good 🙌
